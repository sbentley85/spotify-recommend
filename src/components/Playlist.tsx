import React from "react";

const Playlist = (props: any) => {
	return (
		<div onClick={(e) => props.selectPlaylist(e, props.playlist.id)}>
			{props.playlist.name}
		</div>
	);
};

export default Playlist;
