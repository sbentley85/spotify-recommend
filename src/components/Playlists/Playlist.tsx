import React from "react";
import { IPlaylist } from "../Choices/Choices";
import { IPlaylistCallback } from "../Selection/Selection";
import playlistStyles from "../Tracks/tracks.module.css";

const Playlist = (props: {
	playlist: IPlaylist;
	selectPlaylist: IPlaylistCallback;
}) => {
	return (
		<div
			onClick={(e) => props.selectPlaylist(e, props.playlist)}
			className={playlistStyles.playlist}
		>
			<div
				className={playlistStyles.playlistImage}
				style={
					props.playlist.img
						? { backgroundImage: `url(${props.playlist.img.url})` }
						: { backgroundColor: "black" }
				}
			></div>
			<div className={playlistStyles.playlistName}>
				{props.playlist.name}
			</div>
		</div>
	);
};

export default Playlist;
