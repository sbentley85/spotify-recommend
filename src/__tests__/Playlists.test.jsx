import React from "react";
import Playlist from "../components/Playlists/Playlist";
import { shallow } from "enzyme";


const samplePlaylist = {
	id: "7vsDiGS6XunxvcgFjtp7p6",
	name: "stuff to dl",
	length: 204,
	img: {
		height: 60,
		url:
			"https://mosaic.scdn.co/60/ab67616d0000b273308d4353d4ab62712468a5ebab67616d0000b273636a52ebf89cab3ac84578afab67616d0000b273d37568e85d09fb84ea31959cab67616d0000b273e27f0acabbd9001e0df0af34",
		width: 60,
	},
};

const samplePlaylists = [
	{
		id: "7vsDiGS6XunxvcgFjtp7p6",
		name: "stuff to dl",
		length: 204,
		img: {
			height: 60,
			url:
				"https://mosaic.scdn.co/60/ab67616d0000b273308d4353d4ab62712468a5ebab67616d0000b273636a52ebf89cab3ac84578afab67616d0000b273d37568e85d09fb84ea31959cab67616d0000b273e27f0acabbd9001e0df0af34",
			width: 60,
		},
	},
	{
		id: "7vsDiGS6XunxvcgFjtp7p6",
		name: "stuff to dl",
		length: 204,
		img: {
			height: 60,
			url:
				"https://mosaic.scdn.co/60/ab67616d0000b273308d4353d4ab62712468a5ebab67616d0000b273636a52ebf89cab3ac84578afab67616d0000b273d37568e85d09fb84ea31959cab67616d0000b273e27f0acabbd9001e0df0af34",
			width: 60,
		},
	},
];

describe("Playlist component", () => {
	const clickFn = jest.fn();
	it("renders playlist without crashing", () => {
		shallow(
			<Playlist playlist={samplePlaylist} selectPlaylist={clickFn} />
		);
	});

	it("renders playlist details from props", () => {
		const myPlaylist = shallow(
			<Playlist playlist={samplePlaylist} selectPlaylist={clickFn} />
		);
		expect(myPlaylist).toMatchSnapshot();
	});
});

