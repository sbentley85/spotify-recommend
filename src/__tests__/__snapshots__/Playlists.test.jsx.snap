// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playlist component renders playlist details from props 1`] = `
<div
  className="playlist"
  onClick={[Function]}
>
  <div
    className="playlistImage"
    style={
      Object {
        "backgroundImage": "url(https://mosaic.scdn.co/60/ab67616d0000b273308d4353d4ab62712468a5ebab67616d0000b273636a52ebf89cab3ac84578afab67616d0000b273d37568e85d09fb84ea31959cab67616d0000b273e27f0acabbd9001e0df0af34)",
      }
    }
  />
  <div
    className="playlistName"
  >
    stuff to dl
  </div>
</div>
`;
